### PR TITLE
gimp: update to 2.10.0

### PR DESCRIPTION
**Pull request by kartik-ynwa**
gimp: update to 2.10.0
gegl3: update to 0.3.34
New package: gegl4-0.4.0
New package: libmypaint-1.3.0
New package: mypaint-brushes-1.3.0

### CODE REVIEW

File: srcpkgs/gegl3/template [hunk 1]
```diff
@@ -16,7 +16,7 @@ maintainer="Juan RP <xtraeme@voidlinux.eu>"
 license="GPL-3, LGPL-3"
```
Comment (maxice8) [inline srcpkgs/gegl3/template]: use SDPX identifiers

File: srcpkgs/gegl4/template [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+configure_args="--disable-docs --without-sdl
+ --without-libavformat --without-graphviz --without-lua --without-libspiro
+ --without-umfpack --with-exiv2 --without-libv4l"
+hostmakedepends="pkg-config intltool"
+makedepends="libglib-devel libpng-devel babl-devel jasper-devel
+ libjpeg-turbo-devel librsvg-devel gtk+-devel exiv2-devel json-glib-devel
+ libraw-devel"
+short_desc="Graph based image processing framework (v0.3)"
+maintainer="Kartik S <kartik.ynwa@gmail.com>"
+license="GPL-3, LGPL-3"
```
Comment (maxice8) [inline srcpkgs/gegl4/template]: use SPDX identifiers

File: srcpkgs/gimp/template [hunk 3]
```diff
@@ -1,51 +1,53 @@
...
+	/etc/gimp/2.0/gimprc
+	/etc/gimp/2.0/gtkrc
+	/etc/gimp/2.0/templaterc
+	/etc/gimp/2.0/unitrc
+	/etc/gimp/2.0/sessionrc
+	/etc/gimp/2.0/controllerrc
+	/etc/gimp/2.0/menurc"
+maintainer="Antonio Malcolm <antonio@antoniomalcolm.com>"
+homepage="http://www.gimp.org"
 license="GPL-3"
```
Comment (maxice8) [inline srcpkgs/gimp/template]: use SPDX identifier

File: srcpkgs/gimp/template [hunk 4]
```diff
@@ -1,51 +1,53 @@
...
 distfiles="https://download.gimp.org/pub/gimp/v${version%.*}/gimp-${version}.tar.bz2"
-checksum=9187a35cc52b110d78124d7b27b68a68ade14a794c2721314bac6134d2a5638a
+checksum=7fcc96fb88cb0a0595d2610f63a15dec245bb37bf9db527d37a24fb75e547de2
+create_wrksrc="yes"
```
Comment (maxice8) [inline srcpkgs/gimp/template]: don't need to be quoted and should be after revision

File: srcpkgs/mypaint-brushes/template [hunk 5]
```diff
@@ -0,0 +1,20 @@
+# Template file for 'mypaint-brushes'
+pkgname=mypaint-brushes
+version=1.3.0
+revision=1
+build_style=gnu-configure
+hostmakedepends="automake"
+short_desc="Brushes used by MyPaint and other software using libmypaint"
+maintainer="Kartik Singh <kartik.ynwa@gmail.com>"
+license="CC0-1, GPL-2, GPL-3"
```
Comment (maxice8) [inline srcpkgs/mypaint-brushes/template]: use SPDX identifiers

**Comment by jnbr** [comment]
It should be fine to update gegl to version 0.4 instead of having an extra package.

**Comment by kartik-ynwa** [comment]
@jnbr GIMP needs gegl 0.2 and 0.4. If something is updated it could be gegl3 but I didn't know if it broke stuff or not.

**Comment by kartik-ynwa** [comment]
I pushed the suggested changes and cleaned the commit log.

Let me know what is thought about gegl[3,4].

**Comment by kartik-ynwa** [comment]
My build of GIMP didn't detect mypaint-brushes even though they were installed. I am looking into this.

**Comment by kartik-ynwa** [comment]
There are two issies with this PR. Any help is appreciated regarding these.

1. Travis CI fails to build. This is unusual because of the error messages it gives. It says `checking for intltool >= 0.40.1...  found` and then `configure: error: Your intltool is too old. You need intltool 0.40.1 or later.`
2. I have added `mypaint-brushes` to both `makedepends` and `depends` because it required at both compile-time and run-time. But after building GIMP, when I do `xi gimp` in my repo, `mypaint-brushes` is not installed automatically. [SCREENSHOT](http://img.ziggi.org/3lhZGnWJ.png).

Any pointers are appreciated.

**Comment by jnbr** [comment]
Entry for libmypaint in common/shlibs is missing:
`=> WARNING: libmypaint-1.3.0_1: libmypaint-1.3.so.0 not found in common/shlibs!`

File: srcpkgs/gimp/template [hunk 6]
```diff
@@ -1,51 +1,54 @@
...
+hostmakedepends="pkg-config intltool gettext-devel"
+makedepends="atk-devel babl-devel json-glib-devel gegl-devel gegl4-devel gtk+-devel glib-devel
+ dbus-glib-devel libgexiv2-devel gdk-pixbuf-devel freetype-devel fontconfig-devel harfbuzz-devel
+ cairo-devel pango-devel lcms-devel libjpeg-turbo-devel jasper-devel librsvg-devel tiff-devel
+ python-devel pygtk-devel zlib-devel bzip2-devel liblzma-devel poppler-glib-devel poppler-data
+ libmng-devel ghostscript-devel libwmf-devel aalib-devel libXcursor-devel libXpm-devel
+ libopenexr-devel alsa-lib-devel webkitgtk2-devel libgudev-devel libmypaint-devel mypaint-brushes
+ aalib-devel"
+depends="desktop-file-utils hicolor-icon-theme iso-codes gegl4-devel libgimp-${version}_${revision}
+ darktable mypaint-brushes"
```
Comment (jnbr) [inline srcpkgs/gimp/template]: no need to depend on darktable, please remove it from the dependencies

**Comment by jnbr** [comment]
> @jnbr GIMP needs gegl 0.2 and 0.4. If something is updated it could be gegl3 but I didn't know if it broke stuff or not.

I fail to see why. Gimp's configure script cheks for `gegl` to be in the path. This is a problem, if the binary is renamed to `gegl4` and might be the reason why you need gegl-0.2.
If you just update the gegl template to version 0.4 without any renaming, it works.

File: srcpkgs/gimp/template [hunk 7]
```diff
@@ -1,51 +1,54 @@
...
+provides="gimp-${version}_${revision}"
```
Comment (jnbr) [inline srcpkgs/gimp/template]: this is not needed. Remove it. Same for all other occurances of "provides" in this template.

File: srcpkgs/gimp/template [hunk 8]
```diff
@@ -1,40 +1,38 @@
...
-license="GPL-3"
+conf_files="
+	/etc/gimp/2.0/gimprc
+	/etc/gimp/2.0/gtkrc
+	/etc/gimp/2.0/templaterc
+	/etc/gimp/2.0/unitrc
+	/etc/gimp/2.0/sessionrc
+	/etc/gimp/2.0/controllerrc
+	/etc/gimp/2.0/menurc"
+maintainer="Antonio Malcolm <antonio@antoniomalcolm.com>"
```
Comment (jnbr) [inline srcpkgs/gimp/template]: Forgive me if I'm wrong but isn't this someone else, unrelated to this template?

Comment (kartik-ynwa) [reply srcpkgs/gimp/template]: Yes but not entirely.He posted the GIMP 2.9.4 template on the forum that this template is based on.

Regardless I have put my deets there.

**Comment by kartik-ynwa** [comment]
@jnbr I have done everything you suggested. My bad for that. GIMP compiles fine with gegl updated to 0.4.0. I have removed gegl4 from my branch.

**Comment by wpbirney** [comment]
@kartik-ynwa awesome! you added the aalib dependency, thanks!

**Comment by kartik-ynwa** [comment]
@wpbirney I don'r know how to check whether it has been actually added or now. If you have any feedback feel free to post it here.

**Comment by kartik-ynwa** [comment]
It's kinda problematic for now, I wil PR for dependencies first, then reopen this.

**Pull request closed without merging**